File: sorting/sort.py
#!/usr/bin/env python3
import random
import sys

def quicksort(arr):
    if len(arr) <= 1:
        return arr
    pivot = arr[len(arr) // 2]
    left = [x for x in arr if x < pivot]
    middle = [x for x in arr if x == pivot]
    right = [x for x in arr if x > pivot]
    return quicksort(left) + middle + quicksort(right)

def main():
    # Generate large random array
    size = 1000000  # 1 million elements
    random.seed(42)  # For reproducible results
    arr = [random.randint(1, 1000000) for _ in range(size)]
    
    # Sort the array
    sorted_arr = quicksort(arr)
    
    # Print first and last few elements to verify
    print(f"First 10: {sorted_arr[:10]}")
    print(f"Last 10: {sorted_arr[-10:]}")

if __name__ == "__main__":
    main()

File: sorting/sort.go
package main

import (
	"fmt"
	"math/rand"
)

func quicksort(arr []int) []int {
	if len(arr) <= 1 {
		return arr
	}

	pivot := arr[len(arr)/2]
	var left, middle, right []int

	for _, x := range arr {
		if x < pivot {
			left = append(left, x)
		} else if x > pivot {
			right = append(right, x)
		} else {
			middle = append(middle, x)
		}
	}

	left = quicksort(left)
	right = quicksort(right)

	return append(append(left, middle...), right...)
}

func main() {
	size := 1000000
	rand.Seed(42)

	arr := make([]int, size)
	for i := 0; i < size; i++ {
		arr[i] = rand.Intn(1000000) + 1
	}

	sorted := quicksort(arr)

	fmt.Print("First 10: ")
	for i := 0; i < 10; i++ {
		fmt.Printf("%d ", sorted[i])
	}
	fmt.Println()

	fmt.Print("Last 10: ")
	for i := size - 10; i < size; i++ {
		fmt.Printf("%d ", sorted[i])
	}
	fmt.Println()
}
